src/ui/map.js

```diff
@@ -133,6 +133,10 @@ class Map extends Camera {
     constructor(options) {
         options = util.extend({}, defaultOptions, options);
 
+        if (options.minZoom && options.maxZoom && options.minZoom > options.maxZoom) {
```

Comment:
There's one case that's slipping through the cracks here:

`maxZoom: 0, minZoom: 1`

This could be fixed by changing the test to `options.minZoom != null && options.maxZoom != null`